File: sdkwork-app-sdk-python/sdkwork_app_sdk/api/departments.py
from typing import Any, Dict, List, Optional
from ..http_client import HttpClient
from ..models import PlusApiResultListDepartmentDetailVO, PlusApiResultListDepartmentVO

class DepartmentsApi:
    """departments API client."""
    
    def __init__(self, client: HttpClient):
        self._client = client

    def get_departments_by_org(self, orgId: str) -> PlusApiResultListDepartmentVO:
        """获取组织的部门列表"""
        return self._client.get(f"/app/v3/api/organization/{orgId}/departments")

    def get_department_tree(self, orgId: str) -> PlusApiResultListDepartmentDetailVO:
        """获取部门树"""
        return self._client.get(f"/app/v3/api/organization/{orgId}/departments/tree")

File: sdkwork-app-sdk-flutter/lib/src/api/departments.dart
import '../http/client.dart';
import '../models.dart';

class DepartmentsApi {
  final HttpClient _client;
  
  DepartmentsApi(this._client);

  /// 获取组织的部门列表
  Future<PlusApiResultListDepartmentVO?> getDepartmentsByOrg(String orgId) async {
    final response = await _client.get(ApiPaths.appPath('/organization/${orgId}/departments'));
    return response is PlusApiResultListDepartmentVO ? response : null;
  }

  /// 获取部门树
  Future<PlusApiResultListDepartmentDetailVO?> getDepartmentTree(String orgId) async {
    final response = await _client.get(ApiPaths.appPath('/organization/${orgId}/departments/tree'));
    return response is PlusApiResultListDepartmentDetailVO ? response : null;
  }
}
